File: EasyMeasure/ResultsViewController.m
//
//  ResultsViewController.m
//  Measure
//
//  Created by Jack on 7/12/13.
//  Copyright (c) 2013 CosmicByte. All rights reserved.
//

#import "ResultsViewController.h"

@interface ResultsViewController ()

@end

@implementation ResultsViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    double distance = self.distance;
    self.distanceLabel.text = [[NSString alloc] initWithFormat:@"approximately %.2f %@", distance, [[NSUserDefaults standardUserDefaults] boolForKey:@"metric"] ? @"m." : @"yds."];

    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"voiceover"] && [AVSpeechSynthesizer class]) {
        AVSpeechUtterance *utterance = [AVSpeechUtterance speechUtteranceWithString:[NSString stringWithFormat:@"The distance measured was approximately %.02f %@.", distance, [[NSUserDefaults standardUserDefaults] boolForKey:@"metric"] ? @"m." : @"yds."]];
        [utterance setRate:0.35];
        AVSpeechSynthesizer *synth = [[AVSpeechSynthesizer alloc] init];
        [synth speakUtterance:utterance];
    }
}

- (void)sendDistance:(double)distance {
    self.distance = distance;
}

- (IBAction)doneButton:(id)sender {
    [self.navigationController popToRootViewControllerAnimated:YES];
}

@end

File: Easy Measure/views/ResultsViewController.swift
//
//  ResultsViewController.swift
//  Easy Measure
//
//  Created by Jack Cook on 6/4/14.
//  Copyright (c) 2014 CosmicByte. All rights reserved.
//

import UIKit
import AVFoundation

class ResultsViewController: UIViewController {

    @IBOutlet var distanceLabel: UILabel!
    @IBOutlet var doneButton: UIButton!
    
    var distance: Double = 0.0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        distanceLabel.text = "approximately \(distance) " + (defaults.boolForKey(metric) ? "m." : "yds.") + "."
        
        if defaults.boolForKey("voiceover") {
            var utterance: AVSpeechUtterance = AVSpeechUtterance(string: "The distance measured was approximately \(distance) " + (defaults.boolForKey(metric) ? "meters." : "yards."))
            utterance.rate = 0.02
            var synthesizer = AVSpeechSynthesizer()
            synthesizer.speakUtterance(utterance)
        }
    }
    
    @IBAction func doneButton(sender: AnyObject) {
        self.navigationController?.popToRootViewControllerAnimated(true)
    }
}
